client/webserver/http.go

```diff
@@ -145,6 +148,81 @@ func (s *WebServer) handleWallets(w http.ResponseWriter, r *http.Request) {
 	s.sendTemplate(w, "wallets", data)
 }
 
+// handleWalletLogFile is the handler for the '/wallets/logfile' page request.
+func (s *WebServer) handleWalletLogFile(w http.ResponseWriter, r *http.Request) {
+	wf, ok := w.(http.Flusher)
+	if !ok {
+		log.Errorf("unable to flush streamed data")
+		http.Error(w, "unable to flush streamed data", http.StatusBadRequest)
+		return
+	}
+
+	err := r.ParseForm()
+	if err != nil {
+		log.Errorf("error parsing form for wallet log file: %v", err)
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		return
+	}
+
+	assetIDQueryString := r.Form["assetid"]
+	if len(assetIDQueryString) != 1 || len(assetIDQueryString[0]) == 0 {
+		log.Error("could not find asset id in query string")
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		return
+	}
+
+	assetID, err := strconv.ParseUint(assetIDQueryString[0], 10, 32)
+	if err != nil {
+		log.Errorf("failed to parse asset id query string %v", err)
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		return
+	}
+
+	logFilePath, err := s.core.GetWalletLogFilePath(uint32(assetID))
+	if err != nil {
+		log.Errorf("failed to get log file path %v", err)
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		return
+	}
+
+	logFile, err := os.Open(logFilePath)
+	if err != nil {
+		log.Errorf("error opening log file: %v", err)
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	assetName := dex.BipIDSymbol(uint32(assetID))
+	logFileName := fmt.Sprintf("dcrdex-%s-wallet.log", assetName)
+	w.Header().Set("Content-Disposition", "attachment; filename="+logFileName)
```

Comment:
If it weren't for the file potentially being written to concurrently I'd say we could even set the content length.  I wonder how this will work out if there's an exclusive lock on the log file.  Normally it's fine to read it though.

Reply:
I think an exclusive lock that doesn't allow reads would be counterproductive for a log file.